server/routes/user_preferences.py

```
@@ -100,12 +100,14 @@ def get_batch_preferences(user_id):
         conn = connect_to_db_as_user()
         cursor = conn.cursor()
-        set_rls_context(cursor, conn, user_id, log_prefix="[UserPrefs:get]")
+        org_id = set_rls_context(cursor, conn, user_id, log_prefix="[UserPrefs:get]")
 
         placeholders = ','.join(['%s'] * len(keys))
+        pref_col = "org_id" if org_id else "user_id"
+        pref_val = org_id if org_id else user_id
         cursor.execute(
-            f"SELECT preference_key, preference_value FROM user_preferences WHERE user_id = %s AND preference_key IN ({placeholders})",
-            [user_id] + keys
+            f"SELECT preference_key, preference_value FROM user_preferences WHERE {pref_col} = %s AND preference_key IN ({placeholders})",
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)